front/app/containers/IdeasNewPage/WithJSONForm/Heading.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import clHistory from 'utils/cl-router/history';
+
+import messages from '../messages';
+
+import GoBackButton from 'containers/IdeasShow/GoBackButton';
+import {
+  Box,
+  Title,
+  Text,
+  stylingConsts,
+  useBreakpoint,
+} from '@citizenlab/cl2-component-library';
+import Button from 'components/UI/Button';
+import Modal from 'components/UI/Modal';
+import { FormattedMessage } from 'utils/cl-intl';
+import { colors } from 'utils/styleUtils';
+import { IProjectData } from 'services/projects';
+import { useSearchParams } from 'react-router-dom';
+
+type Props = {
+  project: IProjectData;
+  titleText: string;
+  isSurvey: boolean;
+  canUserEditProject: boolean;
+};
+
+export const Heading = ({
+  project,
+  titleText,
+  canUserEditProject,
+  isSurvey,
+}: Props) => {
+  const [searchParams] = useSearchParams();
+  const phaseId = searchParams.get('phase_id');
+  const linkToSurveyBuilder = phaseId
+    ? `/admin/projects/${project.id}/phases/${phaseId}/native-survey/edit`
+    : `/admin/projects/${project.id}/native-survey/edit`;
+  const canEditSurvey = canUserEditProject && isSurvey;
+  const isSmallerThanXlPhone = useBreakpoint('phone');
+  const showEditSurveyButton = !isSmallerThanXlPhone && canEditSurvey;
+  const [showLeaveModal, setShowLeaveModal] = useState(false);
+  const openModal = () => {
+    setShowLeaveModal(true);
+  };
+  const closeModal = () => {
+    setShowLeaveModal(false);
+  };
+  const isSurveyOnMobile = isSmallerThanXlPhone && isSurvey;
+
+  return (
+    <>
+      <Box
+        width="100%"
+        display="flex"
+        flexDirection={isSmallerThanXlPhone ? 'row-reverse' : 'column'}
+        justifyContent="center"
+        alignItems="center"
+        background={isSurveyOnMobile ? colors.primary : undefined}
+        pt={isSmallerThanXlPhone ? '0px' : '60px'}
+        pb={isSmallerThanXlPhone ? '0px' : '40px'}
+        mb={isSmallerThanXlPhone ? '20px' : '0px'}
+        height={isSmallerThanXlPhone ? '64px' : undefined}
+        position={isSmallerThanXlPhone ? 'fixed' : undefined}
+        top={isSmallerThanXlPhone ? '0px' : undefined}
+        zIndex="3"
```

Comment:
I am curious. What are the z index values that we use in the application? Do we have rules about the z index values, or can developers choose the values?

Reply:
At the moment, the developers choose the values. It would be nice to standardise though 🤔